index.html

```diff
@@ -1,390 +1,1451 @@
+        [=this=].{{[[writable]]}}.
+      <li>If [=this=].{{[[state]]}} is not `"opened"`, return `null`.
+      <li>If [=this=].{{[[writeFatal]]}} is `true`, return `null`.
+      <li>Let |stream:WritableStream| be a [=new=] {{WritableStream}}.
+      <li>Let |writeAlgorithm| be the following steps:
+        <ol>
+          <li>Let |promise:Promise| be [=a new promise=].
+          <li>
+            [=In parallel=], run the following steps:
+            <ol>
+              <li>Invoke the operating system to write |chunk| to the port.
+              <li>
+                If the chunk was successfully written [=queue a task=] to
+                [=resolve=] |promise| with `undefined` on the [=serial port task
+                source=].
+              <li>
+                If an operating system error was encountered, [=reject=]
+                |promise| with an {{UnknownError}}.
+              <li>
+                If the port was disconnected, [=queue a task=] to [=reject=]
+                |promise| with a {{NetworkError}} on the [=serial port task
+                source=], set [=this=].{{[[writeFatal]]}} to `true`, and invoke
+                the steps to [=handle closing the writable stream=].
+            </ol>
+          <li>Return |promise|.
+        </ol>
+
+        <div class="note">
+          [[STREAMS]] specifies that |writeAlgorithm| will only be invoked after
+          the {{Promise}} returned by a previous invocation of this algorithm
+          has resolved. This is not ideal as it prevents implementations from
+          coalescing multiple chunks waiting in the {{WritableStream}}'s
+          internal queue into a single request to the operating system.
+
+          <p>
+          An alternative specification of this algorithm would define a software
+          buffer in which bytes are explicitly queued for transmission so that
+          multiple requests can be coalesced.
+        </div>
+
+      <li>
+        Let |abortAlgorithm| be the following steps:
+        <ol>
+          <li>Let |promise| be [=a new promise=].
+          <li>Run the following steps [=in parallel=].
+            <ol>
+              <li>
+                Invoke the operating system to discard the contents of all
+                software and hardware transmit buffers for the port.
+              <li>Invoke the steps to [=handle closing the writable stream=].
+              <li>
+                [=Queue a task=] to resolve |promise| on the [=serial port task
+                source=].
+            </ol>
+          <li>Return |promise|.
+        </ol>
+
+        <p class="note">
+          [[STREAMS]] specifies that |abortAlgorithm| will only be invoked after
+          the {{Promise}} returned by a previous invocation of |writeAlgorithm|
+          (if any) has resolved. This blocks abort on completion of the most
+          recent write operation. This could be fixed by passing an
+          {{AbortSignal}} to |writeAlgorithm|.
```

Comment:
Perhaps cross-reference https://github.com/whatwg/streams/issues/1015 to make it clearer you're talking about a potential enhancement to the streams API.